enclave/sgx/eeid_attester.c

```diff
@@ -0,0 +1,186 @@
+
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <stdlib.h>
+#include <string.h>
+
+#include <openenclave/enclave.h>
+
+#include <openenclave/attestation/sgx/eeid_attester.h>
+#include <openenclave/bits/attestation.h>
+#include <openenclave/bits/eeid.h>
+#include <openenclave/internal/raise.h>
+#include <openenclave/internal/sgx/eeid_plugin.h>
+#include <openenclave/internal/sgx/plugin.h>
+#include <openenclave/internal/trace.h>
+
+extern const void* __oe_get_eeid();
+
+#include <openenclave/attestation/sgx/attester.h>
+#include <openenclave/enclave.h>
+
+static oe_result_t _eeid_attester_on_register(
+    oe_attestation_role_t* context,
+    const void* config_data,
+    size_t config_data_size)
+{
+    OE_UNUSED(context);
+    OE_UNUSED(config_data);
+    OE_UNUSED(config_data_size);
+
+    oe_result_t result = OE_UNEXPECTED;
+
+    oe_attester_t* sgx_attest = oe_sgx_plugin_attester();
+    result = oe_register_attester(sgx_attest, NULL, 0);
+    if (result != OE_ALREADY_EXISTS)
+        return result;
+
+    return result;
```

Comment:
Not sure about the logic here. Whether the if condition satisfies, the same result is returned. Why would you do the if check?

Reply:
Done.